src/node_jsrtapi.cc

```diff
@@ -315,8 +320,42 @@ napi_status napi_create_constructor(napi_env e, const char* utf8name, napi_callb
   CHECK_JSRT(JsCreatePropertyIdUtf8("constructor", 12, &pid));
   CHECK_JSRT(JsSetProperty(prototype, pid, constructor, false));
 
+  int instancePropertyCount = 0;
+  int staticPropertyCount = 0;
   for (int i = 0; i < property_count; i++) {
-    CHECK_NAPI(napi_define_property(e, reinterpret_cast<napi_value>(prototype), &properties[i]));
+    if ((properties[i].attributes & napi_static_property) != 0) {
+      staticPropertyCount++;
+    }
+    else {
+      instancePropertyCount++;
+    }
+  }
+
+  std::vector<napi_property_descriptor> descriptors;
+  descriptors.reserve(instancePropertyCount > staticPropertyCount ?
```

Comment:
Nit: since we have an stl dependency anyway, can we use max here - makes the code just a tad bit more readable